cfme/tests/infrastructure/test_host.py

```diff
@@ -312,3 +315,30 @@ def _finalize():
             actual_state, power_state
         ),
     )
+
+
+@test_requirements.infra_hosts
+@pytest.mark.meta(coverage=[1738664])
+@pytest.mark.parametrize(
+    "report_format", ["Download as Text", "Download as CSV", "Print or export as PDF"]
+)
+def test_infrastructure_hosts_bz_1738664(
+    appliance, setup_provider, provider, report_format
+):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: low
+        initialEstimate: 1/3h
+    Bugzilla:
+        1738664
+
+    """
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    hosts_view.toolbar.download.item_select(report_format)
+    if report_format == "Print or export as PDF":
+        handle_extra_tabs(hosts_view)
+    hosts_view.navigation.select("Compute")
```

Comment:
Rather than navigating in this manner, why not navigate using the provider object?
```python,
try:
    navigate_to(provider, "Details", wait_for_view=10)
except TimedOutError:
    pytest.fail("Couldn't navigate to provider details page after downloading report") 
```

Or will this break the test in some way that I'm not seeing?

Reply:
Your last comment is correct. The bug is not reproduced unless I navigate as hosts_view.navigation.select("Compute") or something else that actually clicks on "Compute".

Reply:
@prichard77 in that case I would put a `wait` in the `create_view` and then nest things in a `try/except` block that fails after `except TimedOutError`

Reply:
@prichard77 I would still expect this to be nested in a `try/except` block. The `create_view` will also need a `wait` because, otherwise the test will fail at L424

Reply:
I am having a little difficulty seeing why and where I need to put things in a try/except block. I see create_view and is_displayed used in several places without try/except. I have run the test several times against builds where it is supposed to fail (not fixed) and where it should pass. For my testing it passes and fails in the correct runs. I am not trying to argue, just to understand. Can you explain further? Is my test at risk of failing incorrectly 
, but I have not seen it because I am getting lucky with the timing or load on the system?

Reply:
Spoke offline, since @prichard77 is not using the `wait` kwarg, this implementation is :+1: